### ISSUE TITLE
Clarification: Is file offset specific to fd or an open file handle?

### ISSUE THREAD

**Issue by yagehu** [WebAssembly/WASI#366]
In Linux, each open file has an offset and multiple file descriptors can point to the same open file handle in the system-wide open file table. So it can look like this:

```
File descriptor table         System-wide open file table
+-----+                       +--------+-----------+
|   # |                       | offset | inode ptr |
+=====+                       +========+===========+
|   0 |------+--------------> |      0 |       ... |
+-----+      |                +--------+-----------+
| ... |      |
+-----+      |
|  42 |------+
+-----+
```

My question is: Should this still be the case in WASI? The WASI [comment](https://github.com/WebAssembly/WASI/blob/7256722d5d194be36842ba65d9983dff3718d87d/phases/ephemeral/witx/wasi_ephemeral_fd.witx#L246) for `fd_tell` says:

```
  ;;; Return the current offset of a file descriptor.
  ;;; Note: This is similar to `lseek(fd, 0, SEEK_CUR)` in POSIX.
```

which to me implies offset is specific to each FD.

The [Linux man page for `lseek`](https://www.man7.org/linux/man-pages/man2/lseek.2.html) instead says:

> lseek() repositions the file offset of the open file description associated with the file descriptor

which makes it clear that offset is not associated with a file descriptor but with the open file description/handle.

**Comment by sunfishcode** [comment]
Indeed, WASI basically doesn't have a concept of open file descriptions. This is actually one of the reasons why WASI doesn't have `dup` or any other way to request multiple file descriptors for the same file description. The main use case for `dup` involves `fork` which [we're not supporting](https://github.com/WebAssembly/WASI/blob/master/docs/DesignPrinciples.md#relationship-to-posix) anyway, so we chose to keep our options open.
